lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -161,6 +161,21 @@ IntegerAttr circt::firrtl::getIntOnesAttr(Type type) {
   return getIntAttr(type, APInt(width, -1));
 }
 
+/// Return the single assignment to a Property value. It is assumed that the
+/// single assigment invariant is enforced elsewhere.
+PropAssignOp circt::firrtl::getPropertyAssignment(Value value) {
```

Comment:
Instead of assert, consider making the expected type part of the contract, via `TypedValue<PropertyType>()`, possibly adding a `using` alias if that seems likely to have use elsewhere (see `FIRRTLValue` and `FIRRTLBaseValue`).

Reply:
Yeah, good call, I will switch to that! I need to get in the habit of using TypedValue more.